packages/neuron-ui/src/widgets/Tabs/index.tsx

```diff
@@ -49,23 +53,41 @@ const DefaultVariant = <T extends any = {}>({
   )
 }
 
-export const Tabs = <T extends any = {}>({ tabs, Variant, onChange, ...rest }: TabsProps<T>) => {
+export const Tabs = <T extends {}, E extends {} = DefaultVariantProps>({
+  tabs,
+  Variant = DefaultVariant<T>,
+  onTabChange,
+  ...rest
+}: {
+  tabs: Tab<T>[]
+  Variant?: (props: VariantProps<T> & E) => ReactElement<any, any> | null
+  onTabChange?: (currentTab: Tab<T>) => void
+  // The finer-grained type of E is lost during destructuring, so a manual exclusion operation is performed to address this.
+} & Omit<E, keyof VariantProps<T>>) => {
   if (tabs.length === 0) {
     throw new Error('Tabs must have at least one tab')
   }
 
-  const [selectedTabId, setSelectedTabId] = useState<string | number>(tabs[0].id)
+  const [selectedTabId, setSelectedTabId] = useState<TabId>(tabs[0].id)
   const selectedTab = tabs.find(tab => tab.id === selectedTabId) ?? tabs[0]
 
-  const handleTabChange = (id: string | number) => {
-    setSelectedTabId(id)
-    onChange?.(id)
-  }
+  const combinedOnTabChange = useCallback<VariantProps<T>['onTabChange']>(
+    id => {
+      setSelectedTabId(id)
+      const selectedTab = tabs.find(tab => tab.id === id) ?? tabs[0]
+      onTabChange?.(selectedTab)
```

Comment:
Would it be better to just pass the id? Or we can pass the entire object through the second argument.

Reply:
In the callback, it is possible to destructure `id`, so I think it should be fine.

Reply:
> In the callback, it is possible to destructure `id`, so I think it should be fine.

I mean, in most situations, users only care about the id they get, but the entire object.

Reply:
I'm not entirely sure if that's the case because regular users usually don't need to be concerned about tab changes.
When they focus on a specific tab, they may often require additional information from the tab. This would require writing an additional computation. 
I feel that the current implementation doesn't incur significant additional cost, which is why it was implemented this way.